elements/rh-subnav/rh-subnav.ts

```diff
@@ -0,0 +1,188 @@
+
+import styles from './rh-subnav.css';
+
+/**
+ * Subnav
+ * @slot - Place element content here
+ */
+@customElement('rh-subnav')
+export class RhSubnav extends LitElement {
+  static readonly styles = [styles];
+
+  static isLink(element: HTMLAnchorElement): element is HTMLAnchorElement {
+    return element instanceof HTMLAnchorElement;
+  }
+
+  @queryAssignedElements() private links!: HTMLAnchorElement[];
+
+  @query('[part="links"]') private linkList!: HTMLElement;
+
+  #_allLinks: HTMLAnchorElement[] = [];
+
+  #showScrollButtons = false;
+
+  #overflowOnLeft = false;
+
+  #overflowOnRight = false;
+
+  #scrollTimeout?: ReturnType<typeof setTimeout>;
+
+  #rovingTabindexController = new RovingTabindexController(this);
+
+  /**
+   * Sets color theme based on parent context
+   */
+  @colorContextConsumer() private on?: ColorTheme;
+
+  /**
+   * Sets color palette, which affects the element's styles as well as descendants' color theme.
+   * Overrides parent color context.
+   * Your theme will influence these colors so check there first if you are seeing inconsistencies.
+   * See [CSS Custom Properties](#css-custom-properties) for default values
+   */
+  @colorContextProvider()
+  @property({ reflect: true, attribute: 'color-palette' }) colorPalette?: ColorPalette;
+
+  get #allLinks() {
+    return this.#_allLinks;
+  }
+
+  set #allLinks(links: HTMLAnchorElement[]) {
+    this.#_allLinks = links.filter(link => (this.constructor as typeof RhSubnav).isLink(link));
+  }
+
+  /** Time in milliseconds to debounce between scroll events and updating scroll button state */
+  protected static readonly scrollTimeoutDelay: number = 0;
+  /** Icon name to use for the scroll left button */
+  protected static readonly scrollIconLeft: string = 'angle-left';
+  /** Icon name to use for the scroll right button */
+  protected static readonly scrollIconRight: string = 'angle-right';
+  /** Icon set to use for the scroll buttons */
+  protected static readonly scrollIconSet: string = 'fas';
+
+  async connectedCallback() {
+    super.connectedCallback();
+    await this.updateComplete;
+    this.#setOverflowState();
+    window.addEventListener('resize', () => {
+      this.#onScroll();
+    }, { capture: false });
+  }
+
+  render() {
+    const { scrollIconSet, scrollIconLeft, scrollIconRight } = this.constructor as typeof RhSubnav;
+    const { on = '' } = this;
+    return html`
+      <nav part="container" class="${classMap({ [on]: !!on })}">${!this.#showScrollButtons ? '' : html`
+        <button id="previousLink"
+            aria-label="${this.getAttribute('label-scroll-left') ?? 'Scroll left'}"
+            ?disabled="${!this.#overflowOnLeft}"
+            @click="${this.#scrollLeft}">
+          <pf-icon icon="${scrollIconLeft}" set="${scrollIconSet}" loading="eager"></pf-icon>
+        </button>`}
+        <slot part="links" @slotchange="${this.#onSlotchange}"></slot> ${!this.#showScrollButtons ? '' : html`
+        <button id="nextLink"
```

Comment:
Since scrolling can be done via keyboard, by using arrow keys to get to an item, give this a `tabindex=-1` so that keyboard users don't have to waste keystrokes on it. And since this is really about item visibility, we could just make it `aria-hidden` and remove `aria-label`.

Reply:
Good catch I thought that I had added those at one point.

Reply:
I'll have to upstream `aria-hidden` and remove `aria-label` changes to fix in tabs.